src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/DeserializationHelpers.cs

```diff
@@ -94,6 +94,18 @@ internal static class DeserializationHelpers
             IODataInstanceAnnotationContainer instanceAnnotationContainer = GetAnnotationContainer(propertyInfo, resource);
 
             SetInstanceAnnotations(oDataResource, instanceAnnotationContainer, deserializerProvider, readContext);
+
+            SetInstanceAnnotationsForEdmChangedObject(resource, instanceAnnotationContainer);
+        }
+
+        private static void SetInstanceAnnotationsForEdmChangedObject(object resource, IODataInstanceAnnotationContainer instanceAnnotationContainer)
```

Comment:
>SetInstanceAnnotationsForEdmChangedObject [](start = 28, length = 41)

We should be able to treat an EdmChangedObject just like any other object by attempting to cast to IODataInstanceAnnotationContainer in GetAnnotationContainer.  Having a separate method for setting instance annotations on a changed object is a red flag that our factoring is not correct.

Reply:
This method (now) differentiates between transient and persistent annotations.

---
In reply to: [556076758](https://github.com/OData/WebApi/pull/2275#discussion_r556076758) [](ancestors = 556076758)